### DISCUSSION TITLE
External Editor Option For Mobile

### DISCUSSION THREAD

**Discussion by justlimeguy** [godotengine/godot-proposals#11065] [Editor]
### Describe the project you are working on

working on a game In mobile version of godot cuz my phone is more powerful than my PC.

### Describe the problem or limitation you are having in your project

The external Programs Option On Android Editor Opens file manager which is not the way we set on android.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

This is a proposal to implement the mobile based "app selection menu" for the mobile editors of godot.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

will help in editing files on spot as godot 4.3 as of now retains editor data on exit until it's quit from venue.

### If this enhancement will not be used often, can it be worked around with a few lines of script?

it would be helpful for mobile developers who work on mobile.
if all version of godot are compile from same code then might just add an exception for when it is compiled for mobile editors.

### Is there a reason why this should be core and not an add-on in the asset library?

it is an essential feature for the android editor.

**Comment by Norrox** [comment]
i get what your saying, i'm just curious, is it common to use an external editor on mobile ??

**Comment by justlimeguy** [comment]
> i get what your saying, i'm just curious, is it common to use an external editor on mobile ??

yes ,always.
even while in desktop usage devs usually load the files from the project directory into external nal programs to do live changes and changes in general.

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

**Comment by arkology** [comment]
I don't even know which "External Editors" could be on mobile. Especially with custom syntax highlight support.
I think for mobile it will be better solution (for now) to enable sensor orientation mode in distraction free mode.

**Reply by justlimeguy** [reply]
> I don't even know which "External Editors" could be on mobile. Especially with custom syntax highlight support. I think for mobile it will be better solution (for now) to enable sensor orientation mode in distraction free mode.

there are few.
-several 2d pixelart editors
-3d mesh editors [prisma3d and 3d modeling app ]
-audio editors
-svg editor (vectormotion)
there are not that many code editors that support godot but that's not a problem so far, but yes..
if godot can work on having a vertical code editor option for mobile as a replacement for "floating code editor"  on desktop could be more user friendly.

**Reply by arkology** [reply]
> > I don't even know which "External Editors" could be on mobile. Especially with custom syntax highlight support. I think for mobile it will be better solution (for now) to enable sensor orientation mode in distraction free mode.
> 
> there are few. -several 2d pixelart editors -3d mesh editors [prisma3d and 3d modeling app ] -audio editors -svg editor (vectormotion) there are not that many code editors that support godot but that's not a problem so far, but yes.. if godot can work on having a vertical code editor option for mobile as a replacement for "floating code editor" on desktop could be more user friendly.

Oh, I forgot that there are not only external **text** editors, sorry.
